google-cloud-spanner/src/test/java/com/google/cloud/spanner/connection/ConnectionStatementExecutorTest.java

```diff
@@ -237,4 +251,290 @@ public void testStatementSetPgTransactionModeNoOp() {
+      public Date getDate(String columnName) {
+        return null;
+      }
+
+      @Override
+      public boolean[] getBooleanArray(int columnIndex) {
+        return new boolean[0];
+      }
+
+      @Override
+      public boolean[] getBooleanArray(String columnName) {
+        return new boolean[0];
+      }
+
+      @Override
+      public List<Boolean> getBooleanList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<Boolean> getBooleanList(String columnName) {
+        return null;
+      }
+
+      @Override
+      public long[] getLongArray(int columnIndex) {
+        return new long[0];
+      }
+
+      @Override
+      public long[] getLongArray(String columnName) {
+        return new long[0];
+      }
+
+      @Override
+      public List<Long> getLongList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<Long> getLongList(String columnName) {
+        return null;
+      }
+
+      @Override
+      public double[] getDoubleArray(int columnIndex) {
+        return new double[0];
+      }
+
+      @Override
+      public double[] getDoubleArray(String columnName) {
+        return new double[0];
+      }
+
+      @Override
+      public List<Double> getDoubleList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<Double> getDoubleList(String columnName) {
+        return null;
+      }
+
+      @Override
+      public List<BigDecimal> getBigDecimalList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<BigDecimal> getBigDecimalList(String columnName) {
+        return null;
+      }
+
+      @Override
+      public List<String> getStringList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<String> getStringList(String columnName) {
+        return null;
+      }
+
+      @Override
+      public List<ByteArray> getBytesList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<ByteArray> getBytesList(String columnName) {
+        return null;
+      }
+
+      @Override
+      public List<Timestamp> getTimestampList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<Timestamp> getTimestampList(String columnName) {
+        return null;
+      }
+
+      @Override
+      public List<Date> getDateList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<Date> getDateList(String columnName) {
+        return null;
+      }
+
+      @Override
+      public List<Struct> getStructList(int columnIndex) {
+        return null;
+      }
+
+      @Override
+      public List<Struct> getStructList(String columnName) {
+        return null;
+      }
+    };
+    return rs;
+
+  }
+  @Test
+  public void testStatementExplain(){
+    String sql = "verbose select * from table";
+    try {
+      ResultSet rs = subject.statementExplain(sql);
```

Comment:
Use `Assert.assertThrows` to verify that a method invocation throws a specific exception. See here for an example: https://github.com/googleapis/java-spanner/blob/5f156f2efdb4726679766b385d500a030c24e477/google-cloud-spanner/src/test/java/com/google/cloud/spanner/DialectTest.java#L93

Reply:
Done